src/mlpack/methods/preprocess/preprocess_scale_main.cpp

```diff
@@ -0,0 +1,174 @@
+    " or zcawhitening, should be between -1 to 1.", "r", 0.000001);
+
+PARAM_INT_IN("seed", "Random seed (0 for std::time(NULL)).", "s", 0);
+PARAM_INT_IN("min_value", "Starting value of range for min_max_scaler.",
+    "b", 0);
+PARAM_INT_IN("max_value", "Ending value of range for min_max_scaler.",
+    "e", 1);
+PARAM_INT_IN("function", "function to apply,either 0:Transform or 1:Inverse"
+  "Transform", "f", 0)
+// Loading/saving of a model.
+PARAM_MODEL_IN(ScalingModel, "input_model", "Input Scaling model.", "m");
+PARAM_MODEL_OUT(ScalingModel, "output_model", "Output scaling model.", "M");
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  const std::string scalerMethod = CLI::GetParam<string>("scaler_method");
+
+  if (CLI::GetParam<int>("seed") == 0)
+    mlpack::math::RandomSeed(std::time(NULL));
+  else
+    mlpack::math::RandomSeed((size_t) CLI::GetParam<int>("seed"));
+  // Make sure the user specified output filenames.
+  RequireAtLeastOnePassed({ "output", "output_model"}, false,
+      "no output will be saved");
+  // Check scaler method.
+  RequireParamValue<std::string>("scaler_method",
+      [](std::string x) { return x == "standard_scaler" || x ==
+      "min_max_scaler" || x == "mean_normalization" || x == "max_abs_scaler" ||
+      x == "whitening";}, true, "scaler_method must be one among min_max_"
+      "scaler, max_abs_scaler, pcawhitening, standard_scaler, zcawhitening"
+      " or mean_normalization.");
+
+  // Load the data.
+  arma::mat& input = CLI::GetParam<arma::mat>("input");
+  arma::mat output;
+  ScalingModel* m;
+  if (CLI::HasParam("input_model"))
+  {
+    m = CLI::GetParam<ScalingModel*>("input_model");
+  }
+  else
+  {
+    m = new ScalingModel(CLI::GetParam<int>("min_value"),
+        CLI::GetParam<int>("max_value"), CLI::GetParam<double>("epsilon"));
+    if (scalerMethod == "standard_scaler")
+    {
+      m->ScalerType() = ScalingModel::ScalerTypes::STANDARD_SCALER;
+    }
+    else if (scalerMethod == "min_max_scaler")
+    {
+      m->ScalerType() = ScalingModel::ScalerTypes::MIN_MAX_SCALER;
+    }
+    else if (scalerMethod == "max_abs_scaler")
+    {
+      m->ScalerType() = ScalingModel::ScalerTypes::MAX_ABS_SCALER;
+    }
+    else if (scalerMethod == "mean_normalization")
+    {
+      m->ScalerType() = ScalingModel::ScalerTypes::MEAN_NORMALIZATION;
+    }
+    else if (scalerMethod == "zcawhitening")
+    {
+      m->ScalerType() = ScalingModel::ScalerTypes::ZCAWHITENING;
+    }
+    else if (scalerMethod == "pcawhitening")
+    {
+      m->ScalerType() = ScalingModel::ScalerTypes::PCAWHITENING;
+    }
+    m->Fit(input);
+  }
+  if (!CLI::GetParam<int>("function"))
+  {
+    m->Transform(input, output);
+  }
+  else
+  {
+    m->InverseTransform(input, output);
+  }
+
+  // save the output
+  if (CLI::HasParam("output"))
+    CLI::GetParam<arma::mat>("output") = std::move(output);
+
+  CLI::GetParam<ScalingModel*>("output_model") = m;
```

Comment:
I think we should add `if (CLI::HasParam("output_model"))` here; is there any reason why you didn't?  (Maybe I forgot something?)

Reply:
Sorry, I completely missed it.